pkg/oci/driver/driver.go

```diff
@@ -94,15 +109,70 @@ func deriveVolumeOCID(regionKey string, volumeName string) string {
 	return volumeOCID
 }
 
+func constructKubeClient() (*kubernetes.Clientset, error) {
+	// read master url, auth token and root ca cert from disk
+	// which will have been written during daemonset deploy.sh
+	u, err := ioutil.ReadFile(GetDriverDirectory() + "/master_url")
+	if err != nil {
+		return nil, err
+	}
+
+	t, err := ioutil.ReadFile(GetDriverDirectory() + "/token")
+	if err != nil {
+		return nil, err
+	}
+
+	tls := rest.TLSClientConfig{}
+	rootCAFile := GetDriverDirectory() + "/ca.crt"
+	if _, err := certutil.NewPool(rootCAFile); err != nil {
+		return nil, err
+	}
+	tls.CAFile = rootCAFile
+
+	kubeClient, err := kubernetes.NewForConfig(&rest.Config{
+		Host:            strings.TrimSpace(string(u)),
+		BearerToken:     string(t),
+		TLSClientConfig: tls,
+	})
+
+	return kubeClient, err
+}
+
+// findNodeID returns the OCID for the given nodeName
+func findNodeID(k kubernetes.Interface, nodeName string) (string, error) {
+	var id string
+
+	nodeList, err := k.CoreV1().Nodes().List(metav1.ListOptions{})
+	if err != nil {
+		return "", err
+	}
+	for _, node := range nodeList.Items {
+		if node.Name == nodeName {
+			id = node.Spec.ProviderID
```

Comment:
Should we strip the oci prefix that can be on id's here?

Why can't we hit Metadata for the Id